pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -4164,3 +4164,14 @@ query I
 SELECT count(*) FROM child80828
 ----
 0
+
+# See #104546 which was a regression when an invalid column name was specified,
+# when adding a FK reference.
+subtest unknown_column_name
+
+statement ok
+create table t104546 (a int primary key);
+create table t104546_fk_src (a int primary key, b int);
+
+statement error column "b" does not exist
```

Comment:
actually one more nit. could you make this `statement error pgcode 42703 column "b" does not exist` to verify that the correct PG error code is used too?